### DISCUSSION TITLE
Testing a fixture with multiple params, before injecting it to other tests?

### DISCUSSION THREAD

**Discussion by FeryET** [pytest-dev/pytest#9777] [Q&A]
Imagine I have this:

```python

@pytest.mark.parameterize('s',['a', 'b', 'c'])
def test_create_something_from(s):
      return create_something_from(request.param)
@pytest.fixture
def fixture_obj():
    return create_something_from('s')

def test_some_function(fixture_obj):
     # do some tests with the fixture_obj
```

How can I make this:

1. Less verbose (testing a fixture while creating it?)
2. if not possible to directly test a fixture, know that my test for `create_something_from` runs before the creation of the fixture?

**Comment by Northxw** [comment]
I think the following code should be the effect you want to achieve, here I use pytest.fixture for parameterization.

```python
import pytest

def create_something_from(args):
    return args

@pytest.fixture(params=[1, 2, 3])
def fixture_obj(request):
    return create_something_from(request.param)

def test_demo(fixture_obj):
    print("\n", fixture_obj)
```

**Reply by FeryET** [reply]
> I think the following code should be the effect you want to achieve, here I use pytest.fixture for parameterization.
> 
> ```python
> import pytest
> 
> def create_something_from(args):
>     return args
> 
> @pytest.fixture(params=[1, 2, 3])
> def fixture_obj(request):
>     return create_something_from(request.param)
> 
> def test_demo(fixture_obj):
>     print("\n", fixture_obj)
> ```

The problem in my case is that, I want to test that my function which creates the fixtures works or not. Debugging errors in fixture class are way harder. I want to somehow test it, and if it works then move on to fixturizing it.
